dbt/include/databricks/macros/adapters.sql

```diff
@@ -111,6 +111,37 @@
   {% endif %}
 {% endmacro %}
 
+{% macro optimize(relation) %}
+  {{ return(adapter.dispatch('optimize', 'dbt')(relation)) }}
+{% endmacro %}
+
+{% macro databricks__optimize(relation) %}
+  {% if config.get('zorder', False) and config.get('file_format', 'delta') == 'delta' %}
+    {% if var('DATABRICKS_SKIP_OPTIMIZE', 'false')|lower != 'true' and var('databricks_skip_optimize', 'false')|lower != 'true' %}
+      {% call statement('run_optimize_stmt') %}
+        {{ get_optimize_sql(relation) }}
+      {% endcall %}
+    {% endif %}
+  {% endif %}
+{% endmacro %}
+
+{% macro get_optimize_sql(relation) %}
+  {% if config.get('zorder', False) and config.get('file_format', 'delta') == 'delta' %}
+     {%- set zorder = config.get('zorder', none) -%}
+    optimize {{ relation }}
+    {# TODO: predicates here? WHERE ...  #}
+    {% if zorder is sequence and zorder is not string %}
```

Comment:
I would propose moving the check for `zorder is string` to line 139. `zorder is sequence` will not return True if `zorder` is a string. In the case where `zorder is not sequence` and is also not a string, Jinja will try to iterate over it and it will fail.

Reply:
I don't understand. zorder should only be sequence or string. This same sort of jinja code is all over dbt and its macros and it works.